# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/IdentityCredentialStorageService.h
# Commit: 63d9972bab1f
# Full Hash: 63d9972bab1f267a940ec30c7cea7a4c41e54fa6
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing leaks @ mochitest.toml
#   
#   This reverts commit a97cb1d6c08bc38e8644bfc916298b847dbac3a1.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 801e25cb4bfd -r 63d9972bab1f toolkit/components/credentialmanagement/IdentityCredentialStorageService.h
--- a/toolkit/components/credentialmanagement/IdentityCredentialStorageService.h	Thu Jun 05 18:51:36 2025 +0000
+++ b/toolkit/components/credentialmanagement/IdentityCredentialStorageService.h	Thu Jun 05 21:40:49 2025 +0300
@@ -14,6 +14,7 @@
 #include "mozilla/OriginAttributes.h"
 #include "mozIStorageConnection.h"
 #include "mozIStorageFunction.h"
+#include "mozilla/dom/IPCIdentityCredential.h"
 #include "nsIAsyncShutdown.h"
 #include "nsIFile.h"
 #include "nsIIdentityCredentialStorageService.h"
@@ -86,6 +87,8 @@
   // Grab all data from the disk database and insert it into the memory
   // database/ This is used at start up
   nsresult LoadMemoryTableFromDisk();
+  nsresult LoadLightweightMemoryTableFromDisk();
+  nsresult LoadHeavyweightMemoryTableFromDisk();
 
   // Used to (thread-safely) track how many operations have been launched to the
   // worker thread so that we can wait for it to hit zero before close the disk
@@ -119,6 +122,14 @@
 
   static nsresult ClearData(mozIStorageConnection* aDatabaseConnection);
 
+  static nsresult UpsertLightweightData(
+      mozIStorageConnection* aDatabaseConnection,
+      const dom::IPCIdentityCredential& aData);
+
+  static nsresult DeleteLightweightData(
+      mozIStorageConnection* aDatabaseConnection,
+      const dom::IPCIdentityCredential& aData);
+
   static nsresult DeleteDataFromOriginAttributesPattern(
       mozIStorageConnection* aDatabaseConnection,
       OriginAttributesPattern const& aOriginAttributesPattern);